main.cpp

```diff
@@ -310,9 +374,13 @@ int uthread_suspend(int tid) {
  * @param tid - the tid of the thread needing to be terminated
  */
 int uthread_terminate(int tid) {
+    disable_interrupts();
+    
     // Verify that tid is valid
-    if(tid >= num_threads || tid < 0)
+    if(tid >= num_threads || tid < 0) {
+	enable_interrupts();
 	return -1;
+    }
```

Comment:
It has been emphasized in the class that we should minimize the duration of code for which signals are disabled, we could just disable interrupts after we check the tid, right?

Reply:
I was thinking about that too. I was thinking it would improve clarity if I just disabled/enabled at the very start/end of every function, but it might just be better to disable them for the things that are absolutely necessary.